client/src/core/client/framework/lib/errors/relayNetworkRequestError.ts

```diff
@@ -30,6 +38,25 @@ const parseDuplicateEmailError = (
   }
 };
 
+const parseUsernameNotProvidedError = (
+  error: RRNLRequestError
+): UsernameNotProvidedErrorObj | null => {
+  if (!error.res || error.res.status !== 403 || !error.res.text) {
+    return null;
+  }
+
+  try {
+    const json = JSON.parse(error.res.text) as UsernameNotProvidedErrorObj;
+    if (!json || !json.error || json.error.code !== "USERNAME_NOT_PROVIDED") {
+      return null;
+    }
+
+    return json;
+  } catch {
+    return null;
+  }
+};
+
```

Comment:
heh, looks like we're going to have to be generalizing this at some point... so many error object types popping up to be parsed through the response text 😅